src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -679,6 +679,17 @@ bool NonClientIslandWindow::_HandleWindowPosChanging(WINDOWPOS* const windowPos)
         return false;
     }
 
+    const auto windowStyle = GetWindowStyle(_window.get());
+    const auto isMaximized = WI_IsFlagSet(windowStyle, WS_MAXIMIZE);
+    const auto isIconified = WI_IsFlagSet(windowStyle, WS_ICONIC);
+
+    if (_titlebar)
+    {
+        _titlebar.SetWindowVisualState(isMaximized ? winrt::TerminalApp::WindowVisualState::WindowVisualStateMaximized :

```

Comment:
my only complaint is that these lines are _too damn long_. But that's a stylistic choice for someone with split panes in their editor :)